includes/class-content-distribution.php

```diff
@@ -339,6 +339,44 @@ protected static function is_post_unlinked( $post_id ) {
 		return get_post_meta( $post_id, self::POST_UNLINKED_META, true );
 	}
 
+	/**
+	 * Upload the thumbnail for a linked post.
+	 *
+	 * @param int    $post_id       The linked post ID.
+	 * @param string $thumbnail_url The thumbnail URL.
+	 */
+	protected static function upload_thumbnail( $post_id, $thumbnail_url ) {
+		$post_payload         = get_post_meta( $post_id, self::POST_PAYLOAD_META, true );
+		$current_thumbnail_id = get_post_thumbnail_id( $post_id );
+
+		// Bail if the post has a thumbnail and the thumbnail URL is the same.
+		if (
+			$current_thumbnail_id &&
+			$post_payload &&
+			$post_payload['post_data']['thumbnail_url'] === $thumbnail_url
+		) {
+			return;
+		}
+
+		// Delete existing thumbnail attachment.
+		if ( $current_thumbnail_id ) {
+			wp_delete_attachment( $current_thumbnail_id, true );
+		}
+
+		if ( ! function_exists( 'media_sideload_image' ) ) {
+			require_once ABSPATH . 'wp-admin/includes/media.php';
+			require_once ABSPATH . 'wp-admin/includes/file.php';
+			require_once ABSPATH . 'wp-admin/includes/image.php';
+		}
+
+		$attachment_id = media_sideload_image( $thumbnail_url, $post_id, '', 'id' );
+		if ( is_wp_error( $attachment_id ) ) {
+			return;
```

Comment:
A silent error here is probably fine, so this is a question more than a request – but do we have a way to log stuff like this?

Reply:
Good call! We have a Debugger class, added a log in 23b93e649d39e0091f7b28eaabd1b6b7f6ab89c5